lang/java/reef-runtime-multi/src/main/java/org/apache/reef/runtime/multi/client/MultiRuntimeConfigurationBuilder.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.multi.client;
+
+import org.apache.reef.runtime.local.client.parameters.MaxNumberOfEvaluators;
+import org.apache.reef.runtime.yarn.client.ExtensibleYarnClientConfiguration;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.formats.ConfigurationModule;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * A builder for Multi Runtime Configuration.
+ */
+public final class MultiRuntimeConfigurationBuilder {
+  private static final Set<String> SUPPORTED_RUNTIMES = new TreeSet<>(Arrays.asList(
+          org.apache.reef.runtime.yarn.driver.RuntimeIdentifier.RUNTIME_NAME,
+          org.apache.reef.runtime.local.driver.RuntimeIdentifier.RUNTIME_NAME));
+  private static final Set<String> SUPPORTED_DEFAULT_RUNTIMES = new TreeSet<>(Arrays.asList(
+          org.apache.reef.runtime.yarn.driver.RuntimeIdentifier.RUNTIME_NAME));
+
+  private Set<String> runtimeNames = new TreeSet<>();
+  private String defaultRuntime = null;
+  private final HashMap<Class, Object> namedParameters = new HashMap<>();
+
+  private void addNamedParameter(final Class namedParameter,
+                                           final Object namedParameterValue) {
+    if (namedParameterValue == null) {
+      throw new IllegalArgumentException("Named parameter value cannot be null");
+    }
+
+    this.namedParameters.put(namedParameter, namedParameterValue);
+  }
+
+  public MultiRuntimeConfigurationBuilder addRuntime(final String runtimeName) {
```

Comment:
Why not `addRuntime(String id, Configuration runtimeConfiguration)`? That way, it would be extensible from the get go. Also, we wouldn't need the APIs specific to the local runtime in this class.